pkg/asset/machines/aws/machines.go

```diff
@@ -98,7 +99,21 @@ func provider(clusterID, clusterName string, platform *aws.Platform, mpool *aws.
 				Values: []string{fmt.Sprintf("%s_%s_sg", clusterName, role)},
 			}},
 		}},
-	}, nil
+	}
+	if role == "master" {
+		providerConfig.PublicIP = pointer.BoolPtr(true)
```

Comment:
Do we want to hard-code this?  It currently matches [this default][1] from among [these choices][2].  But this code is far from that code.  Can we push this setting up into a shared type ([`aws.MachinePool`][3]?) to stay DRY?

[1]: https://github.com/openshift/installer/blob/6a098fd06dd83278ebfc09b7108e49bbfc357600/pkg/tfvars/tfvars.go#L95
[2]: https://github.com/openshift/installer/blob/6a098fd06dd83278ebfc09b7108e49bbfc357600/pkg/tfvars/aws/aws.go#L7-L12
[3]: https://github.com/openshift/installer/blob/6a098fd06dd83278ebfc09b7108e49bbfc357600/pkg/types/aws/machinepool.go#L3-L22

Reply:
i would not expose that in MachinePool and move
https://github.com/openshift/installer/blob/6a098fd06dd83278ebfc09b7108e49bbfc357600/pkg/tfvars/tfvars.go#L95 to be driven by Machines and MachineSets...

Reply:
@abhinavdahiya do you mean that I should make machines.Master a dependency of TerraformVariables and pass in the Machine object to TFVars ?

Reply:
> @abhinavdahiya do you mean that I should make machines.Master a dependency of TerraformVariables and pass in the Machine object to TFVars ?

Yeah, but not in this PR, i don't think we need to do that now.